src/main/java/it/gov/pagopa/pu/workflow/wf/exportfile/wfexportfile/ExportFileWFImpl.java

```diff
@@ -0,0 +1,112 @@
+import it.gov.pagopa.pu.processexecutions.dto.generated.ExportFile.ExportFileTypeEnum;
+import it.gov.pagopa.pu.processexecutions.dto.generated.ExportFileStatus;
+import it.gov.pagopa.pu.workflow.wf.exportfile.activity.ScheduleExportFileExpirationActivity;
+import it.gov.pagopa.pu.workflow.wf.exportfile.config.ExportFileWFConfig;
+import java.time.LocalDate;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+@Slf4j
+@WorkflowImpl(taskQueues = ExportFileWFImpl.TASK_QUEUE_EXPORT_FILE_WF)
+public class ExportFileWFImpl implements ExportFileWF, ApplicationContextAware {
+  public static final String TASK_QUEUE_EXPORT_FILE_WF = "ExportFileWF";
+  public static final String TASK_QUEUE_EXPORT_FILE_LOCAL_ACTIVITY = "ExportFileWF_LOCAL";
+
+  @Value("${schedule.export-file-expiration.days}")
+  private int expirationDays;
+
+  private ExportFileActivity exportFileActivity;
+  private UpdateExportFileStatusActivity updateExportFileStatusActivity;
+  private ScheduleExportFileExpirationActivity scheduleExportFileExpirationActivity;
+
+  /**
+   * Temporal workflow will not allow to use injection in order to avoid <a href="https://docs.temporal.io/workflows#non-deterministic-change">non-deterministic changes</a> due to dynamic reconfiguration.<BR />
+   * Anyway it allows to override ActivityOptions, but actually it's not supporting the override based on the particular workflow.<BR />
+   * In {@link it.gov.pagopa.pu.workflow.config.TemporalWFImplementationCustomizer} we are already setting defaults to all workflows.<BR />
+   * Use this as an example to override based on the particular workflow.
+   */
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    ExportFileWFConfig wfConfig = applicationContext.getBean(ExportFileWFConfig.class);
+    exportFileActivity = wfConfig.buildExportFileActivityStub();
+    updateExportFileStatusActivity = wfConfig.buildUpdateExportFileStatusActivityStub();
+    scheduleExportFileExpirationActivity = wfConfig.buildScheduleExportFileExpirationActivityStub();
+  }
+
+  @Override
+  public void exportFile(Long exportFileId, ExportFileTypeEnum exportFileType) {
+    log.info("Starting export file workflow for exportFileId: {}", exportFileId);
+    updateExportFileStatus(UpdateStatusRequest.builder()
+      .exportFileId(exportFileId)
+      .oldStatus(ExportFileStatus.REQUESTED)
+      .newStatus(ExportFileStatus.PROCESSING)
+      .build());
+    Pair<ExportFileResult,String> exportResultPair = executeExport(exportFileId, exportFileType);
+    ExportFileResult exportFileResult = exportResultPair.getLeft();
+    String errorDescription = exportResultPair.getRight();
```

Comment:
let's try/catch it instead in order to avoid the Pair creation, I think it should be more readable if doing here instead of returning a Pair, whose right value is filled only in case of exception

ps: let's separate through an empty line the sections related to activities invocations
```suggestion
    
    Pair<ExportFileResult,String> exportResultPair = executeExport(exportFileId, exportFileType);
    ExportFileResult exportFileResult = exportResultPair.getLeft();
    String errorDescription = exportResultPair.getRight();
    
```